spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -89,41 +91,28 @@ class ClientWidget(QWidget, SaveHistoryMixin, SpyderWidgetMixin):
                    '# *** Spyder Python Console History Log ***', ]
 
     def __init__(self, parent, id_,
-                 history_filename, config_options,
-                 additional_options, interpreter_versions,
-                 connection_file=None, hostname=None,
+                 config_options,
+                 additional_options,
+                 interpreter_versions,
                  context_menu_actions=(),
                  menu_actions=None,
-                 is_external_kernel=False,
-                 is_spyder_kernel=True,
                  given_name=None,
                  give_focus=True,
                  options_button=None,
-                 show_elapsed_time=False,
-                 reset_warning=True,
-                 ask_before_restart=True,
-                 ask_before_closing=False,
-                 css_path=None,
                  handlers={},
-                 stderr_obj=None,
-                 stdout_obj=None,
-                 fault_obj=None,
                  initial_cwd=None):
         super(ClientWidget, self).__init__(parent)
-        SaveHistoryMixin.__init__(self, history_filename)
+        SaveHistoryMixin.__init__(self, get_conf_path('history.py'))
 
         # --- Init attrs
+        self.kernel = None
```

Comment:
Please move `self.kernel`, `self.hostname`, `self.show_elapsed_time`, and `self.reset_warning` to the `# --- Other attrs` section below because these attributes are not initialized from variables passed to the constructor now.